ext/standard/basic_functions.stub.php

```diff
@@ -79,6 +79,179 @@ function sort(array &$arg, int $sort_flags = SORT_REGULAR): bool {}
+
+function in_array($needle, array $haystack, bool $strict = false): bool {}
+
+/** @return int|string|false */
+function array_search($needle, array $haystack, bool $strict = false) {}
+
+/** @prefer-ref $arg */
+function extract(array &$arg, $extract_type = EXTR_OVERWRITE, string $prefix = null): ?int {}
+
+function compact($var_name, ...$var_names): ?array {}
+
+/** @return array|false */
+function array_fill(int $start_key, int $num, $val) {}
+
+function array_fill_keys(array $keys, $val): array {}
+
+/**
+ * @param int|float|string $low
+ * @param int|float|string $high
+ * @param int|float $step
+ * @return array|false
+ */
+function range($low, $high, $step = 1) {}
+
+function shuffle(array &$arg): bool {}
+
+function array_pop(array &$stack) {}
+
+function array_shift(array &$stack) {}
+
+function array_unshift(array &$stack, ...$vars): int {}
+
+function array_splice(array &$arg, int $offset, int $length = UNKNOWN, $replacement = []): array {}
+
+function array_slice(array $arg, int $offset, ?int $length = null, bool $preserve_keys = false): array {}
+
+function array_merge(array ...$arrays): array {}
+
+function array_merge_recursive(array ...$arrays): array {}
+
+function array_replace(array $arr1, array ...$arrays): array {}
+
+function array_replace_recursive(array $arr1, array ...$arrays): array {}
+
+function array_keys(array $arg, $search_value = UNKNOWN, bool $strict = false): array {}
+
+/** @return int|string|null */
+function array_key_first(array $arg) {}
+
+/** @return int|string|null */
+function array_key_last(array $arg) {}
+
+function array_values(array $arg): array {}
+
+function array_count_values(array $arg): array {}
+
+/**
+ * @param int|string|null $column_key
+ * @param int|string $index_key
+ * @return array|false
+ */
+function array_column(array $arg, $column_key, $index_key = null) {}
+
+function array_reverse(array $input, bool $preserve_keys = false): array {}
+
+/** @return array|false */
+function array_pad(array $arg, int $pad_size, $pad_value) {}
+
+function array_flip(array $arg): array {}
+
+function array_change_key_case(array $input, int $case = CASE_LOWER): array {}
+
+function array_unique(array $arg, int $flags = SORT_STRING): array {}
+
+function array_intersect_key(array $arr1, array $arr2, array ...$arrays): array {}
+
+// tests break if $arr2 is variadic
+function array_intersect_ukey(array $arr1, array $arr2, $callback_key_compare_func): array {}
+
+function array_intersect(array $arr1, array ...$arrays): array {}
+
+// tests break if $arr2 is variadic
+function array_uintersect(array $arr1, array $arr2, $callback_data_compare_func): array {}
+
+function array_intersect_assoc(array $arr1, array ...$arrays): array {}
+
+// tests break if $arr2 is variadic
+function array_uintersect_assoc(array $arr1, array $arr2, $callback_data_compare_func): array {}
+
+function array_intersect_uassoc(array $arr1, array ...$arr2, callable $callback_key_compare_func): array {}
```

Comment:
Illegal signature. We should probably check in gen_stubs that only the last parameter is variadic.

Reply:
Well that's what userland allows but these internal function have the callable argument after the variadic argument.

Reply:
For most of the functions like this I couldn't make `$arr2` variadic anyway since it caused tests to break. I changed all of these to `...$rest` and added a check in gen_stubs that only the last parameter is variadic.